subsys/bluetooth/controller/ll_sw/ull.c

```diff
@@ -984,6 +984,15 @@ uint8_t ll_rx_get(void **node_rx, uint16_t *handle)
 
 	*node_rx = NULL;
 
+#if defined(CONFIG_BT_CONN) || defined(CONFIG_BT_CTLR_ADV_ISO)
+	/* Save the tx_ack FIFO's last index to avoid the value being changed if there were no
+	 * Rx PDUs and we were pre-empted before calling `tx_cmplt_get()`, that may advance the
+	 * first index beyond ack_last value recorded in node_rx enqueued by `ll_rx_put()` call
+	 * when we are in the `else` clause below.
+	 */
+	uint8_t tx_ack_last = mfifo_fifo_tx_ack.l;
```

Comment:
Just so I understand, `mfifo_fifo_tx_ack.l` may be modified somewhere else while we are in this function, and that's why we need to "back it up" at the beginning?

Reply:
The value is updated in `mfifo_fifo_tx_ack.l` and stored in Rx-ed PDU, if any, as `ack_last` corresponding to when a Rx data is received. This is the `if` clause.

If there is no Rx-ed Data PDU, then `mfifo_fifo_tx_ack.l` is to be used to generate Num of Completed Packets before any new Rx Data PDU on air. 

But under race, more Num of Completed Packets before Rx-ed PDU could be generated, and an incorrect/stale `ack_last`  will be consequently be stored in the Rx-ed PDU.